dio/lib/src/breadcrumb_client_adapter.dart

```diff
@@ -0,0 +1,71 @@
+// ignore_for_file: strict_raw_type
+
+import 'dart:typed_data';
+
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+import 'adapter/dio_adapter.dart';
+
+/// A [Dio](https://pub.dev/packages/dio)-package compatible HTTP client adapter
+/// which records requests as breadcrumbs.
+///
+/// Remarks:
+/// If this client is used as a wrapper, a call to close also closes the
+/// given client.
+class BreadcrumbClientAdapter extends HttpClientAdapter {
+  // ignore: public_member_api_docs
+  BreadcrumbClientAdapter({HttpClientAdapter? client, Hub? hub})
+      : _hub = hub ?? HubAdapter(),
+        _client = client ?? createAdapter();
+
+  final HttpClientAdapter _client;
+  final Hub _hub;
+
+  @override
+  Future<ResponseBody> fetch(
+    RequestOptions options,
+    Stream<Uint8List>? requestStream,
+    Future? cancelFuture,
+  ) async {
+    // See https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+
+    var requestHadException = false;
+    int? statusCode;
+    String? reason;
+    int? responseBodySize;
```

Comment:
This is currently unused. @kuhnroyal is it somehow possible to get the response size?